analyzers/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -65,33 +66,33 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKind.ObjectCreationExpression,
                 SyntaxKindEx.ImplicitObjectCreationExpression);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(VerifyRecordDeclaration, SyntaxKindEx.RecordClassDeclaration);
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                VerifyRecordDeclaration,
+                SyntaxKindEx.RecordClassDeclaration,
+                SyntaxKindEx.RecordStructDeclaration);
         }
 
         private static void VerifyMethodDeclaration(SyntaxNodeAnalysisContext context)
         {
             var methodDeclaration = (BaseMethodDeclarationSyntax)context.Node;
             var methodSymbol = context.SemanticModel.GetDeclaredSymbol(methodDeclaration);
-
             if (methodSymbol == null || methodSymbol.IsOverride)
             {
                 return;
             }
 
             VerifyReturnType(context, methodDeclaration, methodSymbol);
-
-            var stringUrlParams = GetStringUrlParamIndexes(methodSymbol);
+            var stringUrlParams = StringUrlParamIndexes(methodSymbol);
             if (!stringUrlParams.Any())
             {
                 return;
             }
```

Comment:
Maybe not for this PR, but in general, there are two checks involved in VerifyReturnType and GetStringUrlParamIndexes

* syntactic: parameter or method name contains URI like text
* semantic: return type or parameter is a string

This could be split, to do the semantical work only after the syntactical work found candidates.

Reply:
The implementation isn't very nice overall. I didn't want to go deeper in this PR